pkg/controller/controller_test.go

```diff
@@ -2456,11 +2459,47 @@ func assertNumEvents(t *testing.T, strings []string, number int) {
 	}
 }
 
+// failfFunc is a type that defines the common signatures of T.Fatalf and
+// T.Errorf.
+type failfFunc func(t *testing.T, msg string, args ...interface{})
+
+func fatalf(t *testing.T, msg string, args ...interface{}) {
+	t.Log(string(debug.Stack()))
+	t.Fatalf(msg, args...)
+}
+
+func errorf(t *testing.T, msg string, args ...interface{}) {
+	t.Log(string(debug.Stack()))
+	t.Errorf(msg, args...)
+}
+
+// assertion and expectation methods:
+//
+// - assertX will call t.Fatalf
+// - expectX will call t.Errorf and return a boolean, allowing you to drive a 'continue'
+//   in a table-type test
+
 func assertNumberOfActions(t *testing.T, actions []clientgotesting.Action, number int) {
+	testNumberOfActions("" /* name */, fatalf, t, actions, number)
+}
+
+func expectNumberOfActions(name string, t *testing.T, actions []clientgotesting.Action, number int) bool {
```

Comment:
That's fair.  I couldn't decide quickly so I just chose one at random.  I'm happy to change it.